guidance/rust_code/src/graph/mod.rs

```diff
@@ -0,0 +1,346 @@
+use std::fs::File;
+use std::io::{prelude::*, BufReader, Seek, SeekFrom};
+
+#[derive(PartialEq, PartialOrd, Debug)]
+pub(self) struct GPSPoint {
+    latitude: f64,
+    longitude: f64,
+}
+
+#[derive(Debug)]
+pub struct Edge {
+    // For debugging, finding out what line we are looking at on the map
+    name: String,
+    gps_points: Vec<GPSPoint>,
+}
+
+#[derive(Debug)]
+pub struct Vertex {
+    // Will be used to display key locations to UI
+    name: String,
+
+    // Will be used to identify adjacent nodes and edges
+    pub(self) gps_point: GPSPoint,
+
+    // Key data to determine the shortest path using Dijkstra's Algorithm
```

Comment:
Are these necessary if the Graph struct performs Dijkstra's algorithm?

Reply:
This is okay. The self keyword means gps_point is only public to modules within the same directory. In this case, the unit tests.